modules/Sources/Dependencies/AddressBookClient/AddressBookLiveKey.swift

```diff
@@ -34,304 +39,208 @@ extension AddressBookClient: DependencyKey {
-        data.append(contentsOf: intToBytes(abContacts.contacts.count))
-
-        // Serialize `contacts`
-        abContacts.contacts.forEach { contact in
-            let serializedContact = serializeContact(contact)
-            data.append(serializedContact)
+    private static func syncContacts(
+        contacts: AddressBookContacts,
+        remoteStorage: RemoteStorageClient,
+        storeAfterSync: Bool = true
+    ) throws -> AddressBookContacts {
+        // Ensure remote contacts are prepared
+        var remoteContacts: AddressBookContacts = .empty
+        var storeData = true
+
+        do {
+            let filenameForEncryptedFile = try AddressBookClient.filenameForEncryptedFile()
+            let encryptedData = try remoteStorage.loadAddressBookContacts(filenameForEncryptedFile)
+            remoteContacts = try AddressBookClient.contactsFrom(encryptedData: encryptedData)
+        } catch RemoteStorageClient.RemoteStorageError.fileDoesntExist {
+            storeData = false
+        } catch RemoteStorageClient.RemoteStorageError.containerURL {
+            storeData = false
+        } catch {
+            throw error
         }
 
-        return data
-    }
-    
-    private static func decryptData(_ data: Data) throws -> AddressBookContacts {
-        @Dependency(\.walletStorage) var walletStorage
-
-        // TODO: str4d
-//        guard let encryptionKeys = try? walletStorage.exportAddressBookEncryptionKeys() else {
-//            throw AddressBookClient.AddressBookClientError.missingEncryptionKey
-//        }
-
-        // here you have the encrypted data from the cloud, the blob
-        // you also have a key from the keychain
-        
-        var offset = 0
-
-        // Deserialize `version`
-        let versionBytes = data.subdata(in: offset..<(offset + MemoryLayout<Int>.size))
-        offset += MemoryLayout<Int>.size
-
-        guard let version = AddressBookClient.bytesToInt(Array(versionBytes)) else {
-            return .empty
-        }
-        
-        guard version == AddressBookContacts.Constants.version else {
-            return .empty
-        }
-
-        // Deserialize `lastUpdated`
-        guard let lastUpdated = AddressBookClient.deserializeDate(from: data, at: &offset) else {
-            return .empty
-        }
-
-        // Deserialize `contactsCount`
-        let contactsCountBytes = data.subdata(in: offset..<(offset + MemoryLayout<Int>.size))
-        offset += MemoryLayout<Int>.size
-
-        guard let contactsCount = AddressBookClient.bytesToInt(Array(contactsCountBytes)) else {
-            return .empty
-        }
-        
-        var contacts: [Contact] = []
-        for _ in 0..<contactsCount {
-            if let contact = AddressBookClient.deserializeContact(from: data, at: &offset) {
-                contacts.append(contact)
-            }
-        }
-        
-        let abContacts = AddressBookContacts(
-            lastUpdated: lastUpdated,
+        // Merge strategy
```

Comment:
This function mixes the merge with the rest of the store logic. I would factor the merge out.

Reply:
Set for the follow up task #1408